### ISSUE TITLE
proposal: logical operators, conditional expressions are conditional and should be reported in `no-conditional-in-test`

### ISSUE THREAD

**Issue by camc314** [vitest-dev/eslint-plugin-vitest#900]
```
expect( foo ?? 1).toBe(1)
```

is conditional, and should be reported.

If `null`/`undefined` are allowed values for `foo` users should write: `expect(foo).toBeOneOf([1, null, undefined]);`

Similarly,
```
expect (foo ? foo : 1).toBe(1)
```

should also be reported

Happy to send a PR.

**Comment by Huliiiiii** [comment]
https://github.com/oxc-project/oxc/issues/22496#issuecomment-4471831959
> I checked the history, and the initial implementation was consistent with jest [vitest-dev/eslint-plugin-vitest#69](https://github.com/vitest-dev/eslint-plugin-vitest/pull/69). The implementation was later updated due to issue [vitest-dev/eslint-plugin-vitest#358](https://github.com/vitest-dev/eslint-plugin-vitest/issues/358) (unable to reproduce using oxlint).
> 
> Therefore, I think this issue can be closed, because the current implementation in oxlint is correct.

The issue is that consistency with jest plugin was removed due to a false positive bug. It should be restored without causing a regression.
